src/lang-go.ts

```diff
@@ -654,18 +656,60 @@ function pathname(url: string): string {
 
 export function activate(ctx: sourcegraph.ExtensionContext): void {
     async function afterActivate(): Promise<void> {
-        const address = sourcegraph.configuration.get<Settings>().get('go.serverUrl')
-        if (address) {
-            await activateUsingWebSockets()
+        if (Math.random() < 0) {
+            const address = sourcegraph.configuration.get<Settings>().get('go.serverUrl')
+            if (address) {
+                await activateUsingWebSockets()
+            } else {
+                activateBasicCodeIntel({
+                    languageID: 'go',
+                    fileExts: ['go'],
+                    definitionPatterns: ['\\b%s(,\\s\\w+)*\\s\\:=', '(var|const)\\s%s\\s'],
+                    commentStyle: {
+                        lineRegex: /\/\/\s?/,
+                    },
+                })(ctx)
+            }
         } else {
-            activateBasicCodeIntel({
-                languageID: 'go',
-                fileExts: ['go'],
-                definitionPatterns: ['\\b%s(,\\s\\w+)*\\s\\:=', '(var|const)\\s%s\\s'],
-                commentStyle: {
-                    lineRegex: /\/\/\s?/,
-                },
-            })(ctx)
+            const token = await getOrTryToCreateAccessToken()
+            if (!token) {
+                // console.log('No token')
+            } else {
+                // tslint:disable-next-line: no-empty
+                const noop = () => {}
+
+                const serverURL = sourcegraph.configuration.get<Settings>().value['go.serverUrl']
+                if (serverURL) {
+                    const client = await lspClient.register({
+                        sourcegraph,
+                        transport: lspClient.webSocketTransport({
+                            serverUrl: serverURL,
+                            logger: {
+                                error: noop,
+                                warn: noop,
+                                info: noop,
+                                log: noop,
+                            },
+                        }),
+                        documentSelector: [{ language: 'myLanguage' }],
```

Comment:
This is probably why the lsp-client doesn't show hovers 😉
```suggestion
                        documentSelector: [{ language: 'go' }],
```

Reply:
Oh duh, hah 😅